py-polars/tests/unit/namespaces/test_datetime.py

```diff
@@ -478,6 +478,36 @@ def test_truncate(
     assert out.dt[-1] == stop
 
 
+@pytest.mark.parametrize("time_unit", ["ms", "us", "ns"])
+def test_truncate_duration(time_unit: TimeUnit) -> None:
+    durations = pl.Series(
+        [timedelta(seconds=21), timedelta(seconds=35), timedelta(seconds=59), None]
+    ).dt.cast_time_unit(time_unit)
+
```

Comment:
does this work with negative durations? could you add a test for them too please?